### ISSUE TITLE
Indexing into l:smart_buffer.path breaks on older vim versions

### ISSUE THREAD

**Issue by sung-kim** [mengelbrecht/lightline-bufferline#96]
Commit https://github.com/mengelbrecht/lightline-bufferline/commit/4ac1dddffecc3d1559d22e93a640b50569042eeb uses ` l:smart_buffer.path[sep:]`, which is OK on one system with vim 8.2, but crashes the plugin for vim 7.4. Untested for other versions.

Reproducible with `:call lightline#bufferline#buffers()`, resulting in:
```
Error detected while processing function lightline#bufferline#buffers..<SNR>24_get_buffer_names..<SNR>24_get_buffer_paths:
(...)
E121: Undefined variable: sep:
E15: Invalid expression: sep != -1 && l:smart_buffer.path[sep:] ==# s:dirsep
``` 

Using the older concatenation syntax `sep . ':'` on lines 213 and 222 fixes the issue for me, PR incoming.

**Comment by mengelbrecht** [comment]
The expression `[sep:]` returns the substring from index `sep` to the end of the string. Concatenating `:` to `sep` is not the same and thus won’t work.

Does it work for you if you add a space between `sep` and `:` (leading to `[sep :]`)?

**Comment by sung-kim** [comment]
You're right, here are some sanity checks added near line 209:
```
if strlen(l:name)
  let l:smart_buffer.path = fnamemodify(l:name, ':p:~:.')
  let sep = strridx(l:smart_buffer.path, s:dirsep)
 
  let dbg1 = l:smart_buffer.path[19:]
  let dbg2 = l:smart_buffer.path[sep . ':']
  let dbg3 = l:smart_buffer.path[sep :]
  echom "sbp: " . l:smart_buffer.path
  echom "dbg0: " . sep
  echom "dbg1: " . dbg1
  echom "dbg2: " . dbg2
  echom "dbg3: " . dbg3
  if sep != -1 && l:smart_buffer.path[sep . ':'] ==# s:dirsep
    let sep = strridx(l:smart_buffer.path, s:dirsep, sep - 1)
    echom "dbg4: " . sep
  endif
```
Which gives:
![Selection_009](https://user-images.githubusercontent.com/4373364/133089244-26c92e2a-3dae-400f-a7d8-71b3c770fffa.png)

So I guess `sep . ':'` makes the `l:smart_buffer.path[sep . ':'] ==# s:dirsep` check evaluate true, and `sep` is always updated provided it's not -1 ... 

Should we proceed with `sep :` then?

**Comment by mengelbrecht** [comment]
Just to make sure, the error you mentioned in your first post is gone if you insert a space between `sep` and `:`?

**Comment by sung-kim** [comment]
Yes, I've been using `sep :` since this morning and it appears to work fine.

**Comment by mengelbrecht** [comment]
Do you mind adjusting your PR to that solution? It seems like the correct fix for this issue.

**Comment by sung-kim** [comment]
Done, thanks!

**Comment by mengelbrecht** [comment]
Fixed by #97
